aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -177,6 +177,34 @@ NestedTensorImpl::NestedTensorImpl(
           construct_offsets(nested_size_tensor))
 {}
 
+NestedTensorImpl::NestedTensorImpl(
+    Storage storage,
+    const at::Tensor& base_tensor,
```

Comment:
interesting that this takes both a `storage` and a `base_tensor`. What's the use case for not using base_tensor's storage()? Or is this done to avoid interpreting `base_tensor` as a NT here?

Reply:
I didn't love this but one was because `base_tensor.storage()` returns a const Storage & and we need an r value for the Tensor impl constructor. Didn't want to use const_cast but maybe with your above question about creating views from nested and non nested we can just get the tensor impl and grab the storage from there

Reply:
does it need an rvalue because we take ownership of the `Storage`?

Reply:
Updating